tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/MicroSecEndTest.java

```diff
@@ -0,0 +1,349 @@
+            }
+        }
+    }
+
+    @Test
+    void testDetectCycles() {
+        var model = Paths.get(location, "sqshq", "sqshq_11")
+                .toString();
+
+        var analysis = buildAnalysis(model);
+        analysis.initializeAnalysis();
+        var flowGraph = analysis.findFlowGraphs();
+        flowGraph.evaluate();
+
+        assertTrue(flowGraph.wasCyclic());
+
+    }
+
+    private void checkCrossTransposeFlowGraphViolations(FlowGraphCollection flowGraph,
+            Map<Integer, List<AbstractTransposeFlowGraph>> violatingTransposeFlowGraphs, Set<Integer> violationsSet) {
+        var numOfTransposeFlowGraphs = flowGraph.getTransposeFlowGraphs()
+                .size();
+        // If all TransposeFlowGraphs violate the constraint, the violation is valid
+        // --> if one TransposeFlowGraph has a logging server constraint is not triggered
+        if (violatingTransposeFlowGraphs.get(9)
+                .size() >= numOfTransposeFlowGraphs) {
+            violationsSet.add(9);
+            violationsSet.add(12);
+        }
+        // If all TransposeFlowGraphs violate the constraint, the violation is valid
+        // --> if one TransposeFlowGraph has a secret manager constraint is not triggered
+        if (violatingTransposeFlowGraphs.get(18)
+                .size() >= numOfTransposeFlowGraphs) {
+            violationsSet.add(18);
+        }
+    }
+
+    private boolean hasNodeWithCharacteristic(AbstractTransposeFlowGraph transposeFlowGraph, String constraintRuleType, String constraintRule) {
+        return transposeFlowGraph.stream()
+                .anyMatch(node -> hasNodeCharacteristic(node, constraintRuleType, constraintRule));
+    }
+
+    private boolean checkDataCharacteristicImplication(AbstractVertex<?> node, String constraintPrequisiteType, String constraintPrequisite,
+            String constraintRuleType, String constraintRule) {
+        var dataCharecteristicsByVariableOfPrequisitType = node.getDataCharacteristicNamesMap(constraintPrequisiteType);
+        var dataCharecteristicsByVariableOfRuleType = node.getDataCharacteristicMap(constraintRuleType);
+
+        for (var variable : dataCharecteristicsByVariableOfPrequisitType.keySet()) {
+            if (dataCharecteristicsByVariableOfPrequisitType.get(variable)
+                    .stream()
+                    .anyMatch(n -> n.equals(constraintPrequisite))) {
+                if (!dataCharecteristicsByVariableOfRuleType.get(variable)
+                        .stream()
+                        .anyMatch(n -> n.getValueName()
+                                .equals(constraintRule)))
+                    return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * An API Gateway or similar facade should exist as a single entry point to the system and perform authorization and
+     * authentication of external requests to avoid external entities directly accessing services.
+     */
+
```

Comment:
Please remove this empty line, so the comment shows up when using this method elsewhere